core/src/main/java/com/netease/arctic/utils/CompatiblePropertyUtil.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.utils;
+
+import com.netease.arctic.table.TableProperties;
+import org.apache.iceberg.util.PropertyUtil;
+
+import java.util.Map;
+
+/**
+ * PropertyUtil compatible with legacy properties
+ */
+public class CompatiblePropertyUtil {
+
+  private CompatiblePropertyUtil() {
+  }
+
+  public static boolean propertyAsBoolean(Map<String, String> properties,
+                                          String property, boolean defaultValue) {
+    return PropertyUtil.propertyAsBoolean(properties, getCompatibleProperty(properties, property), defaultValue);
+  }
+
+  public static double propertyAsDouble(Map<String, String> properties,
+                                        String property, double defaultValue) {
+    return PropertyUtil.propertyAsDouble(properties, getCompatibleProperty(properties, property), defaultValue);
+  }
+
+  public static int propertyAsInt(Map<String, String> properties,
+                                  String property, int defaultValue) {
+    return PropertyUtil.propertyAsInt(properties, getCompatibleProperty(properties, property), defaultValue);
+  }
+
+  public static long propertyAsLong(Map<String, String> properties,
+                                    String property, long defaultValue) {
+    return PropertyUtil.propertyAsLong(properties, getCompatibleProperty(properties, property), defaultValue);
+  }
+
+  public static String propertyAsString(Map<String, String> properties,
+                                        String property, String defaultValue) {
+    return PropertyUtil.propertyAsString(properties, getCompatibleProperty(properties, property), defaultValue);
+  }
+
+  private static String getCompatibleProperty(Map<String, String> properties, String property) {
+    String legacyProperty = getLegacyProperty(property);
+    if (legacyProperty != null && properties.containsKey(legacyProperty) && !properties.containsKey(property)) {
+      return legacyProperty;
+    } else {
+      return property;
+    }
+  }
+
+  private static String getLegacyProperty(String property) {
```

Comment:
I think the legacy properties which are in the [ArcticValidator](https://github.com/NetEase/arctic/blob/master/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/descriptors/ArcticValidator.java#L148) class are not compatible. e.g. `metrics.enable`, `log-store.consistency-guarantee.enable`

Reply:
Ok, I will check and add the compatible use of properties in [ArcticValidator](https://github.com/NetEase/arctic/blob/master/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/descriptors/ArcticValidator.java#L148).

But, does it have to be compatible with legacy `log-store.consistency-guarantee.enable`? It seems to be a session property, the same as `dim-table.enable`.

Reply:
I will open another PR to be compatible with `log-store.consistency-guarantee.enable` and `dim-table.enable`.